crates/engine/tree/src/tree/payload_validator.rs

```diff
@@ -522,11 +527,70 @@ where
         // terminate prewarming task with good state output
         handle.terminate_caching(Some(&output.state));
 
+        // Capture parent hash and ancestor overlays for deferred trie input construction.
+        let (parent_hash, overlay_blocks) = ctx
+            .state()
+            .tree_state
+            .blocks_by_hash(block.parent_hash())
+            .unwrap_or_else(|| (block.parent_hash(), Vec::new()));
+
+        // Create a deferred handle to store the sorted trie data.
+        let deferred_trie_data = DeferredTrieData::pending();
+        let deferred_handle_task = deferred_trie_data.clone();
+        let hashed_state_for_trie = hashed_state;
+        let trie_output_for_trie = trie_output;
+        let overlay_blocks_for_trie = overlay_blocks;
+        let deferred_compute_duration =
+            self.metrics.block_validation.deferred_trie_compute_duration.clone();
+
+        // Defer trie sorting and compute trie input to a background task so the validation hot path
+        // can return once the state root is checked. Consumers (DB writes, overlay
+        // providers, proofs) block on the handle only when they actually need the sorted
+        // trie data.
+        let task = move || {
+            let result = panic::catch_unwind(AssertUnwindSafe(|| {
```

Comment:
having a panic here - the concern is that if something crashes the process, any panic within the blocking tasks leaves the underlying data unset which would cause the consumers of trie data to block forever.